services/subscribers/subscriber.go

```diff
@@ -7,36 +7,40 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"os"
 	"strings"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
 	"github.com/jinzhu/gorm"
 
 	"github.com/mailbadger/app/entities"
 	"github.com/mailbadger/app/storage"
 )
 
-const ActionImport = "import"
-
-type SubscribersImporter interface {
+type Service interface {
 	ImportSubscribersFromFile(ctx context.Context, filename string, userID int64, segments []entities.Segment) error
+	RemoveSubscribersFromFile(ctx context.Context, filename string, userID int64) error
 }
 
-type s3Importer struct {
+type service struct {
 	client s3iface.S3API
+	db     storage.Storage
 }
 
 var (
 	ErrInvalidColumnsNum = errors.New("importer: invalid number of columns")
 	ErrInvalidFormat     = errors.New("importer: csv file not formatted properly")
 )
 
-func NewS3SubscribersImporter(client s3iface.S3API) *s3Importer {
-	return &s3Importer{client}
+func New(client s3iface.S3API, db storage.Storage) *service {
+	return &service{client, db}
 }
 
-func (i *s3Importer) ImportSubscribersFromFile(
+func (s *service) ImportSubscribersFromFile(
 	ctx context.Context,
+	filename string,
```

Comment:
We just use the filename here for the error message? It seems a bit excessive. We can remove it from here since we log the filename higher up anyway.